streamz/core.py

```diff
@@ -532,8 +540,12 @@ def __init__(self, upstream, func, *args, **kwargs):
         Stream.__init__(self, upstream, stream_name=stream_name)
         _global_sinks.add(self)
 
-    def update(self, x, who=None):
+    def update(self, x, futures=None, who=None):
         result = self.func(x, *self.args, **self.kwargs)
+        if futures:
+            for future in futures:
+                if not future.done():
+                    future.set_result(None)
```

Comment:
What happens if there are multiple sinks?

My understanding of this line is that this sink says the datum is finished, even if there are more sinks that need to be processed after this one.

Reply:
Yes, this is a problem.  I have previously commented on how we still need to solve the multiple down-streams and provided a potential solution in the PR

Reply:
In your solution, how do you know if all futures are done?

Reply:
I'm not sure I get your question.  The source will wait on them all.

Reply:
Ok so the source needs to keep a reference to all the futures, including the ones created partway through the pipeline.

Reply:
Yes